File: src/AnalyticsService/Transport/Contracts/Responses/StatisticResponse.cs
using System.Text.Json.Serialization;
using AnalyticsService.Service.Model;

namespace AnalyticsService.Transport.Contracts.Responses;

/// <summary>
/// A class representing a contract of this API related to document batch statistic.
/// </summary>
/// <param name="generalBatchStat"></param>
internal sealed class StatisticResponse(GeneralBatchStat generalBatchStat)
{
    [JsonPropertyName("avg_doc_number")]
    public int AverageNumberOfDocs => generalBatchStat.AvgNumberOfDocuments;

    [JsonPropertyName("success_rate")]
    public float SuccessRate => generalBatchStat.SuccessRate;

    [JsonPropertyName("avg_duration")]
    public TimeSpan AverageDuration => generalBatchStat.AvgDuration;
}

File: src/AnalyticsService.Web.Api/Transport/Contracts/Responses/StatisticResponse.fs
namespace AnalyticsService.Web.Api.Transport.Contracts.Responses

open System.Text.Json.Serialization
open AnalyticsService.Web.Api.Service.Model

/// A class representing a contract of this API related to document batch statistic.
[<Sealed>]
type StatisticResponse(stat: GeneralBatchStat) =
    [<JsonPropertyName("avg_doc_number")>]
    member this.AverageNumberOfDocs with get() = stat.AvgNumberOfDocs
    [<JsonPropertyName("success_rate")>]
    member this.SuccessRate with get() = stat.SuccessRate
    [<JsonPropertyName("avg_duration")>]
    member this.AverageDuration with get() = stat.AvgDuration